spyder/workers/updates.py

```diff
@@ -95,5 +96,7 @@ def start(self):
         except Exception:
             error_msg = _('Unable to check for updates.')
 
-        self.error = error_msg
-        self.sig_ready.emit()
+        # Don't show dialog when starting up spyder and an error occur
+        if not self.startup or error_msg is None:
```

Comment:
> Hmm no, the idea is not to show the dialog when there is an error

I don't agree. The idea (for me) is to not show the dialog if there are errors but only at startup (because it's very annoying). However, if you want to manually check for updates, Spyder should an error message in case it can't do it.